### PR TITLE
Fix dotnet CLI tests for Github actions

### LINKED ISSUE #428: Fix dotnet CLI tests for Github actions
Currently failing with a relative path error, commented out in the workflow YAMLs right now.

### PR DESCRIPTION
**Pull request by deesejohn**
Fixes #428

Change summary:
- Created `src` directory for api project
- Created solution file to relate api and unit test projects
- Refactored unit tests to use TestHost
- Uncommented github workflow for C# Unit Tests
- Added `actions/setup-dotnet@v1` to github workflows to support dotnet v5.0.201

### CODE REVIEW

**Comment by deesejohn** [comment]
@googlebot I signed it!

File: .github/workflows/ci-master.yaml [hunk 1]
```diff
@@ -24,6 +24,9 @@ jobs:
     runs-on: self-hosted
     steps:
     - uses: actions/checkout@v2
+    - uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: '5.0.201'
```
Comment (askmeegs) [inline .github/workflows/ci-master.yaml:29]: Quick question - without this step, would your C# tests not run? Or does this just explicitly define the version needed?

File: src/cartservice/src/appsettings.json [hunk 2]
```diff
@@ -11,5 +11,6 @@
       "EndpointDefaults": {
         "Protocols": "Http2"
       }
-    }
+    },
+    "Urls": "http://*:7070"
```
Comment (askmeegs) [inline src/cartservice/src/appsettings.json:15]: Where is this used?  is it required?

File: src/cartservice/tests/CartServiceTests.cs [hunk 3]
```diff
@@ -90,20 +110,24 @@ public async Task AddItem_ItemExists_Updated()
...
-            await client.EmptyCartAsync(new EmptyCartRequest{ UserId = userId });
+            await client.EmptyCartAsync(new EmptyCartRequest { UserId = userId });
         }
 
         [Fact]
         public async Task AddItem_New_Inserted()
         {
-            string userId = Guid.NewGuid().ToString();
+            // Setup test server and client
+            using var server = await _host.StartAsync();
```
Comment (askmeegs) [inline src/cartservice/tests/CartServiceTests.cs:120]: I notice a few lines here (the server/client setup) are duplicated between tests, do you think it's worth writing a helper function or okay to leave as-is?

**Review by askmeegs — COMMENTED** [review]
Overall looks great! just a few clarifying questions, otherwise LGTM.

File: .github/workflows/ci-master.yaml (hunk 1 above)
Comment (deesejohn) [reply .github/workflows/ci-master.yaml:29]: The runtime was defaulting to 3.1 and needed to be set explicitly for the tests to pass. Maybe the self-hosted instance simply needs the latest runtime?

File: src/cartservice/src/appsettings.json (hunk 2 above)
Comment (deesejohn) [reply src/cartservice/src/appsettings.json:15]: This is not required and was simply added when I was running the application natively and needed to replicate the port set within the Dockerfile here: https://github.com/GoogleCloudPlatform/microservices-demo/blob/a7ad8f7cc7cd15e7db024c7b554b04e0d1fbbe52/src/cartservice/Dockerfile#L32

I will revert this change.

File: src/cartservice/tests/CartServiceTests.cs (hunk 3 above)
Comment (deesejohn) [reply src/cartservice/tests/CartServiceTests.cs:120]: I would suggest keeping it as-is to keep top-to-bottom readability clear. I'd be happy migrate to a private method if preferred.

File: .github/workflows/ci-master.yaml (hunk 1 above)
Comment (askmeegs) [reply .github/workflows/ci-master.yaml:29]: Thanks!

File: src/cartservice/tests/CartServiceTests.cs (hunk 3 above)
Comment (askmeegs) [reply src/cartservice/tests/CartServiceTests.cs:120]: Sounds good!

**Review by askmeegs — APPROVED** [review]
Verified that dotnet tests pass locally. Thanks for contributing this work @deesejohn!

**Pull request merged**